File: 0258-add-digits/0258-add-digits.cpp
class Solution {
public:
    int addDigits(int num) {
        string temp = to_string(num);
        int tem = 0;
        if (temp.size() == 1) return num;
        while(temp.size() != 1) {
            tem = 0;
            for (int i = 0; i < temp.size(); tem += temp[i] - '0', i++);
            temp = to_string(tem);
        }
        return tem;
    }
};

File: 0258-add-digits/0258-add-digits.java
class Solution {
    public int addDig(int num) {
        int sum = 0;
        while (num > 0) {
            sum += num % 10;
            num /= 10;
        }
        return sum;
    }
    public int addDigits(int num) {
        int ans = num;
        while(num > 9) {
            ans = addDig(num);
            num = ans;
        }
        return ans;
    }
}